src/Microsoft.AspNetCore.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -76,9 +93,9 @@ protected virtual bool CanWriteType(Type type)
                         mediaTypes.Add(mediaType);
                     }
                 }
-
-                return mediaTypes;
             }
+
+            return mediaTypes;
```

Comment:
This whole method seems to do wonky things - I think there may have been some bugs lurking here..... The purpose of this is to try and enumerate ALL of the possible content types that could be output by an action.

The action may or may not have produces, and this is where that gets a chance to interact with the formatters.

In order to simplify my life, I'm going to adopt the term **concrete media type** to mean a media type with no wildcards of any sort. The expectation is that prior to now, `SupportedMediaTypes` has only contained **concrete media types**

I think an accurate specification of this method would be:

- If `contentType` is null, then all *concrete media types* should be returned
- If `contentType` is non-null then it came from a produces. It should be a *concrete media type*. It should be returned if it's a match for one of the *concrete media types* or a match for one of the wildcards.